### ISSUE TITLE
use the new popover attribute for popovers 

### ISSUE THREAD

**Issue by NullVoxPopuli** [universal-ember/ember-primitives#265]
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/popover

**Comment by johanrd** [comment]
Yeah, but also: Switching to using the Popover API for the Menu component _to a large degree removes the need for the menu component in the first place._

It would be useful to extract floating and tabster into its own modifiers (see also https://github.com/hokulea/aria-voyager/blob/main/packages/ember-aria-voyager/package/README.md) 

**This would make possible the following beautiful API for a menu component with pure HTML plus element modifiers:**

```hbs
<button popovertarget='my-popover'>Open popover</button>
<menu popover id='my-popover' {{float}} {{tabster}}>
  <button role='menuitem' >my menuitem</button>
</menu>
```

popoverModifier can target the button with `popovertarget='${element.id}'`

```ts 
const float modify(element, ,positionalArguments, namedArguments) {
    const referenceElement = document.querySelector(`[popovertarget='${element.id}']`) // with options to override in positionalArguments
    …
}
```

**Comment by NullVoxPopuli** [comment]
can `popover` attribute'd things be nested? can you have a menu open on a menu? 

or is it that dialogs can nest, and you'd popovers _must_ be the top (no nesting).

In any case, I'm still waiting for Baseline to give the go-ahead 
![image](https://github.com/user-attachments/assets/d6639871-3fd0-446b-ba9b-85f401b53756)

I would still want a component though... because wiring up id / target attributes between elements is error prone, and it's nice to have a component do that for me

> `{{float}} {{tabster}}`

I am a big fan of this. 

For floating, you kinda _have_ to use a component for now though, because you need to anchor the floater to something (usually a button)
- modifier https://github.com/universal-ember/ember-primitives/blob/main/ember-primitives/src/floating-ui/modifier.ts#L36
- component https://github.com/universal-ember/ember-primitives/blob/main/ember-primitives/src/floating-ui/component.gts#L53

These were taken from https://github.com/CrowdStrike/ember-velcro as they're not keeping up with things at a pace I like (and they're cool with it)
See:
- https://github.com/CrowdStrike/ember-velcro/pull/199/files

**In particular, this FloatingUI wrapper still needs to be documented**

Here is an example usage, in the Popover component: https://github.com/universal-ember/ember-primitives/blob/main/ember-primitives/src/components/popover.gts#L216
(it combines button and menu together)[^menu]

Though, I will point out, that these components are usually pretty light. But I find the amount of ceremony involved with plain HTML (and to do it correctly, etc) to often be too much. I would like to write less when I'm working on apps.

[^menu]: Here, where the content is used: https://github.com/universal-ember/ember-primitives/blob/main/ember-primitives/src/components/menu.gts#L171 it's using a Role of menu for the menu -- because for a generic popover, you can't really know if your going to be used as a menu or not (seen here: https://github.com/universal-ember/ember-primitives/blob/main/ember-primitives/src/components/popover.gts#L125 -- but also, It does look like the element is customizable, so this probably needs some exploration)



## `{{tabster}}`

I think this is doable, tbh.

The tabster code in menu.gts is specific to menus, but the tabster modifier can assume that it would be used on multiple kinds of things, and `switch` off the element / role that it's used on.

This would be a great PR, tbh <3

**Comment by johanrd** [comment]
> For floating, you kinda _have_ to use a component for now though, because you need to anchor the floater to something (usually a button)

yes, but also no! :) In the popover api I have great success utilizing that a popover is strictly tied to a button with popovertarget=element.id. See my example above with:
```ts
modifier((element) => {
   const target = document.querySelector(`[popovertarget='${element.id}']`)`  
})
```
In my experience, this makes it possible to not need to wrap it in a component. 🎉

**Comment by johanrd** [comment]
> can `popover` attribute'd things be nested? can you have a menu open on a menu?

Yes, I have success using popovers inside popovers and dialogs. – especially for tooltips this is necessary. Works out of the box for most borwsers, and a good polyfill for older versions.

**Comment by NullVoxPopuli** [comment]
> I have success using popovers inside popovers and dialogs. – especially for tooltips this is necessary. Works out of the box for most browsers, and a good polyfill for older versions.

This is excellent news!

Any word on when Safari will be included in the Baseline? is it worth it to wait? or should we go-ahead and start using native popover?
(which reduces a fair amount of internal complexity, and potentially removes the need for portalling, which fixes some a11y issues / tab-order issues).

> strictly tied to a button with popovertarget=element.id. 

This is the ceremony that I like avoiding.

This project _should_ probably provide the primitives for folks to do what you want to do though -- and I would gladly accept a PR to enable that <3

Also, I should probably add to all the component docs answering "why a component" -- would be good to expose _what_ is abstracted / why the component exists in the first place (also allowing smart folks (such as yourself!) to choose when it makes sense to use the underlying modifier primitives)

**Comment by NullVoxPopuli** [comment]
This is turning out nice 
![image](https://github.com/user-attachments/assets/e3ad2db0-159a-434b-be65-7340c390e32e)

**Comment by NullVoxPopuli** [comment]
anchorTo here (I renamed it): https://ember-primitives.pages.dev/5-floaty-bits/anchor-to.md

**Comment by NullVoxPopuli** [comment]
popover is in baseline 2025!

**Comment by NullVoxPopuli** [comment]
I started some exploration here: https://github.com/universal-ember/ember-primitives/pull/615

I am also playing with a separate modifier similar to the `anchorTo` which wires up the ids and attributes for you.

**Comment by NullVoxPopuli-ai-agent** [comment]
## CSS Anchor Positioning + Popover API: scroll-container flip limitation

While working on #699 (replacing Floating UI with CSS Anchor Positioning), we discovered a fundamental limitation with combining the Popover API and CSS Anchor Positioning for scroll-container-aware flipping.

### The problem

`position-try-fallbacks: flip-block` evaluates overflow against the element's **containing block**:

- For `position: fixed` elements → containing block is the **viewport** (or nearest ancestor with `transform`/`will-change`/`filter`)
- For `position: absolute` elements → containing block is the nearest **positioned ancestor** (`position: relative/absolute/fixed/sticky`)

Neither evaluates against a **scroll container** boundary. The flip only triggers when the popover would overflow the containing block, not when the anchor scrolls near the edge of an `overflow: auto` container.

### What we want

When a popover's anchor is inside a scroll container, the popover should flip when there isn't enough room in the requested direction **within the scroll container's visible area** — the same behavior Floating UI provided via JS.

### What works

- `position-visibility: anchors-visible` IS scroll-container-aware — it hides the popover when the anchor is clipped by any overflow ancestor. So the "hide on scroll out" behavior works.
- `position-try-fallbacks: flip-block` works correctly against the viewport when `position: fixed` is used and no ancestor has `transform`/`will-change`/`filter`/`contain`.

### What doesn't work

- **Top layer (`[popover]`) + scroll-container flip**: Elements in the top layer use the viewport as their containing block. `position-try-fallbacks` can't evaluate against a scroll container.
- **`position: absolute` + `contain: paint` on scroll container**: The scroll container becomes the containing block, but `position-try-fallbacks` evaluates against the **full content box** (including scrolled-out content), not the visible area. So flip never triggers.
- **Nested popovers inside `position: sticky` elements**: `position: sticky` creates a containing block for `position: fixed` descendants, constraining nested popovers to the sticky element's bounds.

### Possible approaches

1. **JS scroll listener** (what Floating UI does): Watch scroll events, measure available space, flip `position-area` manually. Works but adds JS complexity.
2. **Accept viewport-level flip**: Use `position: fixed` + `position-try-fallbacks` for viewport flip. Scroll containers get `position-visibility: anchors-visible` (hide) but no flip.  
3. **Future CSS spec**: The CSS Anchor Positioning spec may evolve to support scroll-container-aware `position-try-fallbacks`. There's been discussion in [CSS WG issues](https://github.com/w3c/csswg-drafts/issues) about this gap.

Currently #699 uses approach 2. If scroll-container flip is required, approach 1 (a `scrollFlip` modifier) is available but adds ~60 lines of JS.
